instrumentation_handlers.go

```diff
@@ -13,13 +13,12 @@ import (
 var applicationGCStats = debug.GCStats{}
 var instrument = health.NewStream()
 
-// SetupInstrumentation handles all the intialisation of the instrumentation handler
-func SetupInstrumentation(enabled bool) {
-	if os.Getenv("TYK_INSTRUMENTATION") == "1" {
-		enabled = true
-	}
-
-	if !enabled {
+// setupInstrumentation handles all the intialisation of the instrumentation handler
+func setupInstrumentation(arguments map[string]interface{}) {
+	switch {
+	case arguments["--log-instrumentation"] == true:
+	case os.Getenv("TYK_INSTRUMENTATION") == "1":
```

Comment:
If we going to make it right, let's also introduce configuration variable.

Reply:
How so? Not all configs have flags, and not all flags have configs. For example, the log level doesn't have a config option - and they're fairly similar flags.

Reply:
bump